src/parsers/perf/perfparser.cpp

```diff
@@ -838,6 +841,24 @@ class PerfParserPrivate : public QObject
             }
         }
 
+        const auto costAggregation = Settings::instance()->costAggregation();
```

Comment:
It's not safe to access the Settings API from a background thread, this is not thread safe. I just fixed the similar issue for the debuginfod urls and pushed it directly. Please fix it similarly here